CvGameCoreDLL_Expansion2/CvCity.cpp

```diff
@@ -9759,26 +9759,24 @@ bool CvCity::IsBuildingLocalResourceValid(BuildingTypes eBuilding, bool bTestVis
 	bool bHasAnyOr = false;
 	bool bNeedAnyOr = false;
 	CvString tempTooltip;
-	for (int iResourceLoop = 0; iResourceLoop < GC.getNumResourceInfos(); iResourceLoop++)
+	for (int iPrereq = 0; iPrereq < GD_INT_GET(NUM_BUILDING_RESOURCE_PREREQS); iPrereq++)
```

Comment:
Stone Works has more than 5 prereqs so it'll break.

Reply:
We should turn Building_LocalResourceAnds and Building_LocalResourceOrs into sets instead of arrays.

Reply:
fixed it